app/utils/graphql.ts

```diff
@@ -27,40 +29,74 @@ const DEFAULT_SWR_OPTIONS: SWRConfiguration = {
 /**
  * Fetcher which interacts with our GraphQL API
  */
-const fetcher = <T>([query, variables, token, hasuraRoleName]: [
+const fetcher = <T>([query, variables, token, hasuraRoleName, schema]: [
   RequestDocument,
   Variables,
   string,
-  string
+  string,
+  ZodSchema<T>
 ]): Promise<T> =>
-  request({
+  request<T>({
     url: ENDPOINT,
     document: query,
     variables,
     requestHeaders: {
       Authorization: `Bearer ${token}`,
       "x-hasura-role": hasuraRoleName,
     },
+  }).then((data) => {
+    return schema.parseAsync(data);
```

Comment:
here's where we actually run the schema validation. if this errors, the error will be available in the `error` property returned by the `useQuery()` hook